sc/src/main/java/uk/ac/ebi/atlas/tsne/TSnePlotServiceDao.java

```diff
@@ -89,4 +89,15 @@ public TSnePlotServiceDao(NamedParameterJdbcTemplate namedParameterJdbcTemplate)
                 Integer.class
         );
     }
+
+    private static final String COUNT_CELLS_BY_EXPERIMENT_ACCESSION =
+            "SELECT COUNT(DISTINCT(cell_id)) FROM scxa_tsne WHERE experiment_accession=:experiment_accession";
+    public Integer fetchCellNumberByExperimentAccession(String experimentAccession) {
```

Comment:
I would change this method name to `fetchNumberOfCellsByExperimentAccession`. `CellNumber` sounds like you are referring to an ID of sorts, rather than the count.